BizzNEST/modesto-bizznest-scheduler
# feat(script, json-convert): fetch interns data, refactor intern table, add filtering and select features

## Changes
1. Refactored script.js to use json_array for intern table population and handling intern pairings.
2. Added filtering functionality for location and department in the intern pool.
3. Implemented "Select All" and "Deselect All" functionality to manage intern pairings.
4. Added clear all functionality to reset selected pairings.
5. Added top and bottom logo
6. Added background logo image
7. Added a results page

## Purpose
This update introduces a more efficient way to handle intern data by fetching it once and reusing it throughout the app, refactors the intern table to be dynamically populated from json_convert.js, and adds key features like filtering by department/location, and selecting/deselecting interns.

## Approach
1. Centralized Data Handling: By creating the json_convert.js module, the interns.json data is only fetched once and converted into a flat array that can be reused across different components of the app. This prevents unnecessary re-fetching and simplifies the data handling process.

2. Added Filtering & Selection: The filtering functionality allows users to narrow down the intern pool by location and department, while the "Select All" and "Deselect All" features simplify intern management for users.

## Testing Steps
1. Open the app and check if the intern table populates correctly upon page load.
2. Verify that filtering interns by location and department works as expected.
3. Select individual interns and verify that they are added to the pairing box.
4. Click "Select All" and check if all interns are selected and added to the pairing box.
8. Click "Deselect All" and verify that all interns are removed from the pairing box.
9. Clear all selected pairings using the "Clear All" button and ensure the pairing list is empty.
10. Check browser developer tools to ensure no errors related to fetching the interns.json file.

## Learning

https://www.w3schools.com/jsref/jsref_obj_array.asp
https://www.w3schools.com/jsref/jsref_obj_json.asp
https://www.tutorialspoint.com/how-to-convert-json-string-to-array-of-json-objects-using-javascript
https://www.reddit.com/r/css/comments/t3am53/pure_css_gradientprogressive_blur_i_used/

Closes #21 #20 #24
